app/src/main/java/io/github/fate_grand_automata/ui/card_priority/Render.kt

```diff
@@ -81,39 +102,129 @@ fun CardPriorityListItem.Render(
-                foregroundColor = Color.WHITE,
-                backgroundColor = when (it.position) {
-                    1 -> R.color.colorArts
-                    2 -> R.color.colorQuick
-                    3 -> R.color.colorBuster
-                    4 -> R.color.colorArtsResist
-                    5 -> R.color.colorQuickResist
-                    else -> R.color.colorBusterResist
-                }.let { res -> context.getColor(res) },
-                text = "  ${it.position}  "
+        titleText = stringResource(id = R.string.p_battle_config_servant_priority),
+        messageText= stringResource(R.string.p_battle_config_instructions_drag_and_drop),
+        items = clonePriorities,
+        itemContent = { item ->
+            ListItem(
+                leadingContent = {
+                    Icon(
+                        imageVector = Icons.Default.DragHandle,
+                        contentDescription = stringResource(id = R.string.p_battle_config_servant_priority)
+                    )
+                },
+                trailingContent = {
+                    // Added Blank icon to make the Text Centered
+                    Icon(
+                        imageVector = Icons.Default.Clear,
+                        contentDescription = null,
+                        tint = Color.Transparent
+                    )
+                },
+                colors = ListItemDefaults.colors(
+                    containerColor = prioritiesItemColor(item),
+                ),
+                modifier = Modifier
+                    .fillMaxWidth(),
+                headlineContent = {
+                    Column(
+                        horizontalAlignment = Alignment.CenterHorizontally,
+                        verticalArrangement = Arrangement.Center,
+                        modifier = Modifier.fillMaxHeight()
+                    ) {
+                        Text(
+                            text = "${item.position}",
+                            textAlign = TextAlign.Center,
+                            modifier = Modifier.fillMaxWidth(),
+                            style = MaterialTheme.typography.bodyMedium,
+                            fontWeight = FontWeight.ExtraBold,
+                            color = MaterialTheme.colorScheme.onPrimaryContainer
+                        )
+                    }
+                }
             )
+        },
+        initialContent = {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+            ) {
+                clonePriorities.forEach { item ->
+                    PrioritiesCardItem(
+                        item = item,
+                        modifier = Modifier.weight(1f)
+                    )
+                }
+            }
+        },
+        onSubmit = {
+            clonePriorities = it
+            onSubmit(clonePriorities.toMutableList())
         }
     )
+}
+
+@Composable
+private fun PrioritiesCardItem(
```

Comment:
The text is not white anymore for the HIGH -> LOW priority and SERVANT PRIORITY items.

Also, in both lists the items got wider, so on my phone the RQ entry is not on the screen anymore. If I drag the items to the left so I can see RQ, I also automatically scroll to wave 2 priority.